SPECIFICATION.md

```diff
@@ -0,0 +1,338 @@
+
+## Schema Definition
+
+### Top-level Fields
+
+**Required Fields:**
+
+| Field | Type | Description |
+|-------|------|-------------|
+| `events` | array | Ordered array of Event objects representing the component's lifecycle events. MUST be chronologically ordered by effective date. |
+
+### Event Object
+
+The base object that represents a discrete lifecycle event. All events share these common fields.
+
+**Required Fields:**
+
+| Field | Type | Description |
+|-------|------|-------------|
+| `type` | string | The type of lifecycle event. MUST be one of the defined Event Types. |
+| `effective` | string | The time when the event takes effect, as an RFC3339-formatted timestamp in UTC (ending in "Z"). |
+| `published` | string | The time when the event was first published, as an RFC3339-formatted timestamp in UTC (ending in "Z"). |
+| `modified` | string | The time when the event was last modified, as an RFC3339-formatted timestamp in UTC (ending in "Z"). |
+
+**Optional Fields:**
+
+| Field | Type | Description |
+|-------|------|-------------|
+| `description` | string | Human-readable description of the event. SHOULD be clear and concise. |
+| `references` | array[string] | List of URLs to supporting documentation. SHOULD point to authoritative sources. |
+
+### Event Types
+
+#### generalAvailability
+*Category: Version Event*
+
+Indicates when a component version is released and available for use.
+
+**Additional Required Fields:**
+- `version`: VERS-formatted single version (e.g., "vers:npm/1.0.0")
+
+Example:
+```json
+{
+  "type": "generalAvailability",
+  "effective": "2023-01-01T00:00:00Z",
+  "version": "vers:npm/1.0.0",
+  "published": "2023-01-01T00:00:00Z",
+  "modified": "2023-01-01T00:00:00Z",
+  "description": "Version 1.0.0 is now generally available",
+  "references": [
+    "https://example.com/release-notes"
+  ]
+}
+```
+
+#### endOfDevelopment
+*Category: Version Event*
+
+Indicates when the manufacturer ceases development of a component or service.
+
+**Additional Required Fields:**
+- `range`: VERS-formatted version range (e.g., "vers:npm/>=1.0.0|<2.0.0")
+
+Example:
+```json
+{
+  "type": "endOfDevelopment",
+  "effective": "2024-01-01T00:00:00Z",
+  "range": "vers:npm/>=1.0.0|<2.0.0",
+  "published": "2023-06-01T00:00:00Z",
+  "modified": "2023-06-01T00:00:00Z",
+  "description": "Development of 1.x versions will cease"
+}
+```
+
+#### endOfSupport
+*Category: Version Event*
+
+Indicates when the manufacturer ceases any and all support of a component or service. This point in time marks a transfer of risk from the manufacturer to the consuming organisation or user of the component or service, encompassing all cybersecurity knowledge and known vulnerabilities, with no further assistance provided by the manufacturer.
```

Comment:
This has probably been discussed before, but if `endOfSupport` means that the manufacturer stops accepting vulnerability reports and will not publish any new CVEs, how do you mark the "end of security updates"? There is a certain number of vulnerability reports, which are "UNSUPPORTED WHEN ASSIGNED". These are published, but the manufacturer will never fix them.

Maybe `endOfSupport` should not exonerate the manufacturer from accepting vulnerability reports, which would be published until `endOfLife` is announced.

Reply:
> how do you mark the "end of security updates"

`endOfSupprt` is the way for doing that since it `indicates when the manufacturer ceases any and all support of a component or service`

`endOfSupprt` should not care whether the manufacturer accepts vulnerabilities reports or not, just that they will not be doing any new releases to fix known vulnerabilities.

Reply:
I am not a native speaker, but from the description I would think that `endOfSupport` also means that vulnerability reports will not be accepted and processed.

Reply:
"end of support" to me definitely sounds like it should be equivalent to, "every maintainer is dead".